src/storage/hummock_sdk/src/slice_transform.rs

```diff
@@ -0,0 +1,417 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(dead_code)]
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use risingwave_common::catalog::ColumnDesc;
+use risingwave_common::types::VIRTUAL_NODE_SIZE;
+use risingwave_common::util::ordered::OrderedRowDeserializer;
+use risingwave_common::util::sort_util::OrderType;
+use risingwave_pb::catalog::Table;
+
+use crate::key::{get_table_id, TABLE_PREFIX_LEN};
+
+trait SliceTransform {
+    fn transfer<'a>(&self, full_key: &'a [u8]) -> &'a [u8];
+}
+
+#[derive(Default)]
+pub struct FullKeySliceTransform;
+
+impl SliceTransform for FullKeySliceTransform {
+    fn transfer<'a>(&self, full_key: &'a [u8]) -> &'a [u8] {
+        full_key
+    }
+}
+
+#[derive(Default)]
+pub struct DummySliceTransform;
+
+impl SliceTransform for DummySliceTransform {
+    fn transfer<'a>(&self, full_key: &'a [u8]) -> &'a [u8] {
+        &full_key[0..0]
+    }
+}
+
+pub struct SchemaSliceTransform {
+    // table_catalog: Table,
+    read_pattern_prefix_column: u32,
```

Comment:
please add more docs here

Reply:
some doc for slice_transform at new commit